pmd-apex/src/test/java/net/sourceforge/pmd/cpd/ApexTokenizerTest.java

```diff
@@ -4,89 +4,57 @@
 
 package net.sourceforge.pmd.cpd;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
-
-import java.io.IOException;
-import java.nio.charset.StandardCharsets;
-import java.util.ArrayList;
-import java.util.List;
 import java.util.Properties;
 
-import org.apache.commons.io.IOUtils;
 import org.junit.Test;
 
-import net.sourceforge.pmd.PMD;
-import net.sourceforge.pmd.cpd.SourceCode.StringCodeLoader;
+import net.sourceforge.pmd.cpd.test.CpdTextComparisonTest;
+
+public class ApexTokenizerTest extends CpdTextComparisonTest {
+
+    public ApexTokenizerTest() {
+        super(".cls");
+    }
+
+    @Override
+    protected String getResourcePrefix() {
+        return "../lang/apex/cpd/testdata";
```

Comment:
Note that the override is necessary because test classes are in the same package as the tested tokenizer (`n.s.pmd.cpd`), while test files are in this new package. For 7.0 it would be best to move all tokenizers into the language-specific package:

https://github.com/pmd/pmd/pull/2457#discussion_r421376124

And these resource prefixes could be removed